src/main/java/uk/gov/hmcts/cmc/claimstore/services/legaladvisor/DocAssemblyClient.java

```diff
@@ -0,0 +1,137 @@
+package uk.gov.hmcts.cmc.claimstore.services.legaladvisor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+import uk.gov.hmcts.cmc.ccd.domain.legaladvisor.CCDHearingCourtType;
+import uk.gov.hmcts.cmc.ccd.domain.legaladvisor.CCDOrderDirection;
+import uk.gov.hmcts.cmc.ccd.domain.legaladvisor.CCDOrderDirectionType;
+import uk.gov.hmcts.cmc.claimstore.idam.models.UserDetails;
+import uk.gov.hmcts.cmc.claimstore.processors.JsonMapper;
+import uk.gov.hmcts.cmc.claimstore.services.UserService;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+
+import java.time.LocalDate;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class DocAssemblyClient {
+    private static final String TEMPLATE_NAME = "CV-CMC-GOR-ENG-0004.docx";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final DocAssemblyApi docAssemblyApi;
+    private final UserService userService;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final JsonMapper jsonMapper;
+
+    @Autowired
+    public DocAssemblyClient(
+        DocAssemblyApi docAssemblyApi,
+        UserService userService,
+        AuthTokenGenerator authTokenGenerator,
+        JsonMapper jsonMapper) {
+        this.docAssemblyApi = docAssemblyApi;
+        this.userService = userService;
+        this.authTokenGenerator = authTokenGenerator;
+        this.jsonMapper = jsonMapper;
+    }
+
+    public DocAssemblyResponse  generateOrder(String authorisation, Map<String, Object> caseData) {
+        String templateId = Base64.getEncoder()
+            .encodeToString(TEMPLATE_NAME.getBytes());
+        UserDetails userDetails = userService.getUserDetails(authorisation);
+        CCDCase ccdCase = jsonMapper.fromMap(caseData, CCDCase.class);
+        DocAssemblyRequest docAssemblyRequest = buildRequest(templateId, ccdCase, userDetails);
+        try {
+            return docAssemblyApi.generateOrder(
+                docAssemblyRequest,
+                authorisation,
+                authTokenGenerator.generate());
+        } catch (Exception e) {
+            logger.error("Error while trying to generate an order with docAssembly");
+            throw new DocumentGenerationFailedException(e);
+        }
+    }
+
+    private DocAssemblyRequest buildRequest(String templateId, CCDCase ccdCase, UserDetails userDetails) {
+        List<CCDOrderDirectionType> directionList = ccdCase.getDirectionList();
```

Comment:
can this method be moved to a common mapping util or something, instead of in DocAssemblyClient